### PR TITLE
Mifare Ultralight/NTAG simulation. Add WRITE and COMPATIBLE_WRITE support

### PR DESCRIPTION
**Pull request by McEloff**
New feature to store data from reader to 'hf 14a sim' Ultralight/NTAG emulator memory

### CODE REVIEW

File: armsrc/iso14443a.h [hunk 1]
```diff
@@ -93,6 +93,14 @@ typedef struct {
...
 #endif
 
+#ifndef CheckCrc14A
+# define	CheckCrc14A(data, len)	check_crc(CRC_14443_A, (data), (len))
+#endif 
+
+#ifndef CheckCrc14B
+# define	CheckCrc14B(data, len)	check_crc(CRC_14443_B, (data), (len))
+#endif 
+
```
Comment (iceman1001) [inline armsrc/iso14443a.h]: CheckCrc14B  isn't used ...

File: common/protocols.h [hunk 2]
```diff
@@ -166,7 +166,7 @@ ISO 7816-4 Basic interindustry commands. For command APDU's.
 #define MIFARE_EV1_SETMODE          0x43
 
 #define MIFARE_ULC_WRITE            0xA2
-//#define MIFARE_ULC__COMP_WRITE      0xA0
+#define MIFARE_ULC__COMP_WRITE      0xA0
```
Comment (iceman1001) [inline common/protocols.h]: spelling?   Double underscore

**Comment by McEloff** [comment]
Ok. Commit changed

**Comment by iceman1001** [comment]
Cool!
Lets see if helps ppl out :)

**Comment by iceman1001** [comment]
And do add a line in changelog

**Comment by McEloff** [comment]
Ok. Added into my temporary branch within pull request #145

**Pull request merged**